packages/react-priority-overflow/src/useOverflowContainer.ts

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { createOverflowManager } from '@fluentui/priority-overflow';
+
+import type {
+  OnUpdateItemVisibility,
+  OnUpdateOverflow,
+  OverflowItemEntry,
+  OverflowManager,
+  ObserveOptions,
+} from '@fluentui/priority-overflow';
+import { useEventCallback, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import { UseOverflowContainerReturn } from './types';
+
+/**
+ *
+ * @param update - Callback when overflow state changes
+ * @param options - Options to configure the overflow container
+ * @returns - ref to attach to an intrinsic HTML element and imperative functions
+ */
+export const useOverflowContainer = <TElement extends HTMLElement>(
+  update: OnUpdateOverflow,
+  options: Omit<ObserveOptions, 'onUpdateOverflow'>,
+): UseOverflowContainerReturn<TElement> => {
+  const { overflowAxis, overflowDirection, padding, minimumVisible, onUpdateItemVisibility } = options;
+  // DOM ref to the overflow container element
+  const containerRef = React.useRef<TElement>(null);
+  const updateOverflowItems = useEventCallback(update);
+  const [overflowManager] = React.useState<OverflowManager>(() => createOverflowManager());
+
+  useIsomorphicLayoutEffect(() => {
+    if (!containerRef.current) {
+      return;
+    }
+
+    overflowManager.observe(containerRef.current, {
+      overflowDirection,
+      overflowAxis,
+      padding,
+      minimumVisible,
+      onUpdateItemVisibility,
+      onUpdateOverflow: updateOverflowItems,
+    });
+
+    return () => {
+      overflowManager.disconnect();
+    };
+  }, [
+    updateOverflowItems,
+    overflowManager,
+    overflowDirection,
+    overflowAxis,
+    padding,
+    minimumVisible,
+    onUpdateItemVisibility,
+  ]);
+
+  const registerItem = React.useCallback(
+    (item: OverflowItemEntry) => {
+      overflowManager.addItem(item);
+
+      return () => {
+        item.element.style.removeProperty('display');
```

Comment:
So when you are removing an overflow item from the overflow manager, there is some cleanup required. I don't think this cleanup should be in the native API, so I put it in the react bindings.

With the data-* it will simply be removing an attribute instead

Reply:
done in [c263ed38](https://github.com/microsoft/fluentui/pull/21979/commits/c263ed3878ce309b8d6081578e228429ff6db048)